### PR TITLE
refactor rdf-validator to avoid System/exit in -main

### PR DESCRIPTION
**Pull request by jennet**

### CODE REVIEW

File: src/rdf_validator/core.clj [hunk 1]
```diff
@@ -24,18 +25,22 @@
...
     (doseq [error errors]
       (println error))
     (println)
     (usage summary)))
 
+(defn- report-exception [^Exception ex]
+  (binding [*out* *err*]
+    (println (.getMessage ex))
+    (.printStackTrace ex)
+    (System/exit 1)))
```
Comment (lkitching) [inline src/rdf_validator/core.clj]: I don't think we need to exit here?

File: src/rdf_validator/core.clj [hunk 2]
```diff
@@ -113,24 +118,33 @@
...
+            test-cases (tc/suite-tests suites suites-to-run)
+            test-reporter (reporting/->ConsoleTestReporter)]
+        (run-test-cases test-cases query-variables endpoint test-reporter))
+      (throw (ex-info "Invalid command line arguments" {:type :invalid-cli-arguments
+                                                        :cli-result result})))))
+
+(defn -main
+  [& args]
+  (try
+    (set-missing-value-formatter! on-missing-query-template-variable)
```
Comment (lkitching) [inline src/rdf_validator/core.clj]: Should this be in `inner-main`? We might want to add options around this behaviour if different clients want to configure it.

Comment (jennet) [reply src/rdf_validator/core.clj]: Should we wait until we reconfigure the rdf-validator to move it there?

Comment (lkitching) [reply src/rdf_validator/core.clj]: ok!

**Pull request merged**